### PR TITLE
Enable trajectory property for box objects

### LINKED ISSUE #108: Trajectory attribute for all box objects
## Description

I would like to be able to access box objects from every frame as a trajectory property, as a list of `garnett.trajectory.Box` objects.

## Proposed solution

Similar to existing trajectory attributes, except not as a NumPy array (just a list).

## Additional context

Many boxes: 🗳 🗃 ☑️ 🥊

### PR DESCRIPTION
**Pull request by lyrivera**
## Description
Thi PR enables accessing the frames' boxes from the trajectory object after loading arrays. 

It is based off of #137 so I'm opening as a draft until #137 is merged and potential merge conflicts are resolved.

## Motivation and Context

Resolves: #108 

## How Has This Been Tested?

I added a test to ensure that the behavior works as expected.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/garnett/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [x] I have updated the [credits](https://github.com/glotzerlab/garnett/blob/master/doc/credits.rst).
- [x] I have updated the [Changelog](https://github.com/glotzerlab/garnett/blob/master/changelog.rst).

### CODE REVIEW

File: garnett/dcdfilereader.py [hunk 1]
```diff
@@ -244,6 +242,8 @@ def load_arrays(self):
                            dtype=TYPE_PROPERTIES['types'])
         typeid = np.asarray([f._typeid for f in self.frames],
                             dtype=PARTICLE_PROPERTIES['typeid'])
+        box = np.asarray([f._box for f in self.frames],
+                         dtype=FRAME_PROPERTIES["box"])
```
Comment (bdice) [inline garnett/dcdfilereader.py]: There is not an official PEP8 recommendation but most people prefer single quotes for dict keys, and it matches what's above.
```suggestion
                         dtype=FRAME_PROPERTIES['box'])
```

File: garnett/trajectory.py [hunk 2]
```diff
@@ -1230,6 +1232,18 @@ def image(self):
         self._assert_arrays_loaded()
         return self._check_nonempty_property('_image')
 
+    @property
+    def box(self):
+        """Access the frame boxes as a NumPy array.
+
+        :returns: frame boxes as an (M) element array
+        :rtype: :class:`numpy.ndarray` (dtype= :class:`~.Box`)
```
Comment (bdice) [inline garnett/trajectory.py]: Is it actually `dtype=Box`? I would have expected `dtype=object` since `Box` isn't a NumPy type.

**Review by bdice — CHANGES_REQUESTED** [review]
One big change request: we need to keep everything "in order." The methods, docs, defaults in `trajectory.py`, etc. should all be consistent about the order in which we list properties. e.g. `box, N, types, type_shapes, typeid, position, ...`. I don't care what the order _is_, but consistency is important to ensure that this code can be understood and edited in the future. Our current approach is to group "frame properties" that are scalars like `box` and `N`, then "type properties" of length `T` like `types` and `type_shapes`, then "particle properties" of length `N` like `typeid`, `position`, ...

File: garnett/trajectory.py (hunk 2 above)
Comment (lyrivera) [reply garnett/trajectory.py]: Right, fixed

File: garnett/trajectory.py [hunk 3]
```diff
@@ -24,9 +24,10 @@
 
 DEFAULT_DTYPE = np.float32
 
-# Scalar properties for a frame
+# Scalar/per-frame properties
 FRAME_PROPERTIES = {
     'N': np.uint,
+    'box': object,
```
Comment (bdice) [inline garnett/trajectory.py]:
```suggestion
    'box': object,
    'N': np.uint,
```

File: tests/test_trajectory.py [hunk 4]
```diff
@@ -325,6 +325,18 @@ def test_image(self):
         except AttributeError:
             pass
 
+    def test_box(self):
```
Comment (bdice) [inline tests/test_trajectory.py]: Move to the top, above other properties.

**Review by bdice — APPROVED** [review]
Approving with a couple more re-orders.

**Pull request merged**